Sources/EduIDExpansion/Controllers/Onboarding/Coordinator/OnboardingCoordinator.swift

```diff
@@ -11,28 +10,71 @@ final class OnboardingCoordinator: CoordinatorType {
     }
     
     func start() {
-        navigationController.setNavigationBarHidden(true, animated: false)
         let landingScreen = LandingPageViewController()
         landingScreen.coordinator = self
         landingScreen.screenType = .landingScreen
         navigationController.pushViewController(landingScreen, animated: false)
+        
+        // show navigation bar buttons if needed
+        showNavigationBarButtonsIfNeeded(screenType: .landingScreen)
     }
     
     //MARK: - start scan screen
+    @objc
     func showScanScreen() {
         let scanViewcontroller = ScanViewController()
         scanViewcontroller.coordinator = self
         scanViewcontroller.navigationItem.leftBarButtonItem?.image = scanViewcontroller.navigationItem.leftBarButtonItem?.image?.withRenderingMode(.alwaysTemplate)
         navigationController.navigationBar.tintColor = .white
         navigationController.setNavigationBarHidden(false, animated: true)
         navigationController.pushViewController(scanViewcontroller, animated: true)
+        
+        // show navigation bar buttons if needed
+        showNavigationBarButtonsIfNeeded(screenType: .scanScreen)
     }
     
     //MARK: - show next screen
-    func showNextScreen(currentType: ScreenType) {
-        guard let nextViewController = currentType.nextViewController(current: currentType) else { return }
+    func showNextScreen(currentScreen: ScreenType) {
+        guard let nextViewController = currentScreen.nextViewController(current: currentScreen) else { return }
         
         (nextViewController as? EduIDBaseViewController)?.coordinator = self
         navigationController.pushViewController(nextViewController, animated: true)
+        
+        // show navigation bar buttons if needed
+        showNavigationBarButtonsIfNeeded(screenType: (navigationController.viewControllers.last as? EduIDBaseViewController)?.screenType ?? .none)
+    }
+    
+    //MARK: - back action
+    @objc
+    func goBack() {
+        navigationController.popViewController(animated: true)
+        
+        // show navigation bar buttons if needed
+        showNavigationBarButtonsIfNeeded(screenType: (navigationController.viewControllers.last as? EduIDBaseViewController)?.screenType ?? .none)
+        
+    }
+    
+    //MARK: - show navigationbar buttons if needed
+    func showNavigationBarButtonsIfNeeded(screenType: ScreenType) {
+        // show scanbutton if needed
+        if screenType.showScanButtonInNavigationBar {
+            navigationController.viewControllers.last?.navigationItem.leftBarButtonItem = UIBarButtonItem(image: .qrLogo.withRenderingMode(.alwaysOriginal), style: .done, target: self, action: #selector(showScanScreen))
+        } else {
+            navigationController.viewControllers.last?.navigationItem.leftBarButtonItem = nil;
```

Comment:
This line isn't needed, as 2 lines below it gets set to something else anyway.

Reply:
I removed the line